### PR TITLE
[Containerapp] `az containerapp create/update`: Update models to support new properties with `--yaml` for api-version 2024-02-02-preview

### PR DESCRIPTION
**Pull request by Greedygre**
…

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Containerapp

File: src/containerapp/azext_containerapp/_sdk_enums.py [hunk 1]
```diff
@@ -335,6 +403,76 @@ class ManagedServiceIdentityType(str, Enum, metaclass=CaseInsensitiveEnumMeta):
...
+    CREATING_REVISION = "CreatingRevision"
+    SUCCEEDED = "Succeeded"
+    CANCELED = "Canceled"
+    REBASE_FAILED = "RebaseFailed"
+    REVISION_CREATION_FAILED = "RevisionCreationFailed"
+    IMAGE_PUSH_PULL_FAILED = "ImagePushPullFailed"
+    MANUALLY_SKIPPED = "ManuallySkipped"
+
+
+class PatchingMode(str, Enum, metaclass=CaseInsensitiveEnumMeta):
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_sdk_enums.py:420]: I think we do not want to expose patch related property to customer? can we remove it?

File: src/containerapp/azext_containerapp/_sdk_enums.py [hunk 2]
```diff
@@ -335,6 +403,76 @@ class ManagedServiceIdentityType(str, Enum, metaclass=CaseInsensitiveEnumMeta):
...
+    require the user to manually apply patches. Disabled mode will stop patch detection and auto
+    patching.
+    """
+
+    AUTOMATIC = "Automatic"
+    MANUAL = "Manual"
+    DISABLED = "Disabled"
+
+
+class PatchType(str, Enum, metaclass=CaseInsensitiveEnumMeta):
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_sdk_enums.py:432]: same for all patch related

File: src/containerapp/azext_containerapp/_sdk_enums.py (hunk 1 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/_sdk_enums.py:420]: OK, I will remove them.

File: src/containerapp/azext_containerapp/_sdk_models.py [hunk 3]
```diff
@@ -2736,12 +2750,17 @@ def __init__(
...
+         e.g. WorkflowApp is a kind of Microsoft.App/ContainerApps type. If supported, the resource
+         provider must validate and persist this value. "workflowapp"
+        :paramtype kind: str or ~azure.mgmt.appcontainers.models.Kind
         :keyword managed_environment_id: Deprecated. Resource ID of the Container App's environment.
         :paramtype managed_environment_id: str
         :keyword environment_id: Resource ID of environment.
         :paramtype environment_id: str
         :keyword workload_profile_name: Workload profile name to pin for container app execution.
         :paramtype workload_profile_name: str
+         ~azure.mgmt.appcontainers.models.ContainerAppPropertiesPatchingConfiguration
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_sdk_models.py:2763]: ?

File: src/containerapp/azext_containerapp/_sdk_models.py [hunk 4]
```diff
@@ -3138,6 +3158,179 @@ def __init__(self, *, port: int, host: Optional[str] = None, **kwargs: Any) -> N
         self.port = port
 
 
+class ContainerAppPropertiesPatchingConfiguration(_serialization.Model):  # pylint: disable=name-too-long
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_sdk_models.py:3161]: ??

**Review by Juliehzl — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**